openapi2beans/pkg/generator/schema2package_test.go

```diff
@@ -0,0 +1,452 @@
+	assert.True(t, enumExists)
+	assertJavaEnumRelatesToSchemaType(t, schemaType, enum)
+}
+
+func TestTranslateSchemaTypesToJavaPackageWithClassWithEnum(t *testing.T) {
+	// Given...
+	possibleValues := map[string]string{
+		"randValue1": "randValue1",
+		"randValue2": "randValue2",
+	}
+	schemaTypeMap := make(map[string]*SchemaType)
+	var enumSchemaType *SchemaType
+	enumSchemaName := "MyEnum"
+	enumOwnProp := NewProperty(enumSchemaName, SCHEMAS_PATH + enumSchemaName, "", "string", possibleValues, enumSchemaType, Cardinality{min: 0, max: 1})
+	enumSchemaType = NewSchemaType(enumSchemaName, "", enumOwnProp, nil)
+	schemaTypeMap["#/components/schemas/MyEnum"] = enumSchemaType
+	var classSchemaType *SchemaType
+	classSchemaName := "MyBean"
+	enumPropName := "beansEnum"
+	propMap := make(map[string]*Property)
+	enumProp := NewProperty(enumPropName, SCHEMAS_PATH + classSchemaName + "/" + enumPropName, "", enumSchemaName, possibleValues, enumSchemaType, enumOwnProp.cardinality)
+	propMap["#/components/schemas/MyBean/beansEnum"] = enumProp
+	classOwnProp := NewProperty(classSchemaName, SCHEMAS_PATH + classSchemaName, "", classSchemaName, nil, classSchemaType, Cardinality{min: 0, max: 1})
+	classSchemaType = NewSchemaType(classSchemaName, "", classOwnProp, propMap)
+	schemaTypeMap[SCHEMAS_PATH  + classSchemaName] = classSchemaType
+
+
+	// When...
+	javaPackage := translateSchemaTypesToJavaPackage(schemaTypeMap, TARGET_JAVA_PACKAGE)
+
+	// Then...
+	enum, enumExists := javaPackage.Enums[enumSchemaName]
+	assert.True(t, enumExists)
+	assertJavaEnumRelatesToSchemaType(t, enumSchemaType, enum)
+
+	class, classExists := javaPackage.Classes[classSchemaName]
+	assert.True(t, classExists)
+	assertJavaClassCorrectlyRelatesToSchemaType(t, classSchemaType, class)
+}
+
+func TestTranslateSchemaTypesToJavaPackageWithClassWithStringConstant(t *testing.T) {
+	// Given...
+	propName1 := "MyConstant"
+	possibleValues := map[string]string{
+		"constVal": "constVal",
+	}
+	property := NewProperty(propName1, "#/components/schemas/MyBean/"+propName1, "", "string", possibleValues, nil, Cardinality{min: 0, max: 1})
+	properties := make(map[string]*Property)
+	properties["#/components/schemas/MyBean/"+propName1] = property
+	var schemaType *SchemaType
+	schemaName := "MyBean"
+	ownProp := NewProperty(schemaName, "#/components/schemas/MyBean", "", "object", nil, schemaType, Cardinality{min: 0, max: 1})
+	schemaType = NewSchemaType(schemaName, "", ownProp, properties)
+	schemaTypeMap := make(map[string]*SchemaType)
+	schemaTypeMap["#/components/schemas/MyBean"] = schemaType
+
+	// When...
+	javaPackage := translateSchemaTypesToJavaPackage(schemaTypeMap, TARGET_JAVA_PACKAGE)
+
+	// Then...
+	class, classExists := javaPackage.Classes[schemaName]
+	assert.True(t, classExists)
+	assert.NotEmpty(t, class.ConstantDataMembers)
+	assertJavaClassCorrectlyRelatesToSchemaType(t, schemaType, class)
+}
+
+func TestConvertToConstName(t *testing.T) {
```

Comment:
Seems that there are lots more combinations of input to be tested.
eg: a single-character name ? A name which starts with a number ?
A name which has underscores in already ? An invalid name ?

I see the implementation uses regexes, which are a rich source of bugs, so whenever I see a regex being used I get interested to see all the combinations of inputs are handled in the tests... same for the other convert functions.